code/modules/antagonists/changeling/powers/fakedeath.dm

```diff
@@ -57,20 +57,31 @@
 		return
 
 	to_chat(user, span_notice("We are ready to revive."))
-	build_all_button_icons(UPDATE_BUTTON_NAME|UPDATE_BUTTON_ICON)
 	chemical_cost = 0
 	revive_ready = TRUE
+	build_all_button_icons(UPDATE_BUTTON_NAME|UPDATE_BUTTON_ICON)
 
 /datum/action/changeling/fakedeath/can_sting(mob/living/user)
-	if(HAS_TRAIT_FROM(user, TRAIT_DEATHCOMA, "changeling") && !revive_ready)
-		user.balloon_alert(user, "already reviving!")
+	if(revive_ready)
+		return ..()
+
+	if(!can_enter_stasis(user))
 		return
-	if(!user.stat && !revive_ready) //Confirmation for living changelings if they want to fake their death
-		switch(tgui_alert(usr,"Are we sure we wish to fake our own death?", "Feign Death", list("Yes", "No")))
-			if("No")
-				return
+	//Confirmation for living changelings if they want to fake their death
+	if(user.stat != DEAD)
+		if(tgui_alert(user, "Are we sure we wish to fake our own death?", "Feign Death", list("Yes", "No")) != "Yes")
+			return
+		if(QDELETED(user) || QDELETED(src) || !can_enter_stasis(user))
+			return
+
 	return ..()
 
+/datum/action/changeling/fakedeath/proc/can_enter_stasis(mob/living/user)
+	if(HAS_TRAIT_FROM(user, TRAIT_DEATHCOMA, "changeling"))
```

Comment:
```suggestion
	if(HAS_TRAIT_FROM(user, TRAIT_DEATHCOMA, CHANGELING_TRAIT))
```

for some reason this wasn't using the define, which is a tad bit silly. let's fix that now (on further inspection, it seems a good solid number of things don't use the define macro, odd

![image](https://user-images.githubusercontent.com/34697715/221693013-3b2da6ac-cf35-4e35-9a3b-d57f1feb14d0.png)